shopfloor_reception/services/reception.py

```diff
@@ -324,91 +342,180 @@ def _scan_document__by_picking(self, barcode):
                 )
             return self._select_picking(reception_pickings)
 
-    def _scan_document__by_product(self, barcode):
-        search = self._actions_for("search")
-        product = search.product_from_scan(barcode)
+    def _scan_document__by_product(self, product, barcode):
         if product:
             return self._select_document_from_product(product)
 
-    def _scan_document__by_packaging(self, barcode):
-        search = self._actions_for("search")
-        packaging = search.packaging_from_scan(barcode)
+    def _scan_document__by_packaging(self, packaging, barcode):
         if packaging:
             return self._select_document_from_packaging(packaging)
 
-    def _scan_document__by_lot(self, barcode):
-        return self._select_document_from_lot(barcode)
+    def _scan_document__by_lot(self, lot, barcode):
+        return self._select_document_from_lot(lot)
```

Comment:
No need to check if there is a lot ? like the other `_scan_document__...` functions ?